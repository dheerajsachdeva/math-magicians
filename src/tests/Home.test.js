import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Home from '../components/Home';

test('should render the home component', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should have two paragraphs', () => {
  render(<Home />);
  const headingElement = screen.getByText(/Welcome to our page!/i);
  expect(headingElement).toBeInTheDocument();
});

test('should have two paragraphs', () => {
  render(<Home />);
  const headingElement = screen.queryByText(/link/i);
  expect(headingElement).not.toBeInTheDocument();
});
