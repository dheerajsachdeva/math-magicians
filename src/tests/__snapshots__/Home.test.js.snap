// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="home"
>
  <p>
    A simple to use calculator with an appealing user interface. A virtual calculator built with React is another good project idea for you to try as a beginner.
  </p>
  <br />
  <p>
    We have seen various calculator apps on our mobile phones, computers and different devices as well. Here we have an input box which takes user input as numbers and then try to do operations like addition, subtraction, division and multiplication etc. Here we will learn to create a web app with all the features of the calculator on UI.
  </p>
</div>
`;
